consensus/istanbul/backend/api.go

```diff
@@ -42,6 +42,21 @@ type API struct {
 }
 
 // getHeaderByNumber retrieves the header requested block or current if unspecified.
+func (api *API) getHeaderByNumber(number *rpc.BlockNumber) (*types.Header, error) {
+	if number == nil || *number == rpc.LatestBlockNumber || *number == rpc.PendingBlockNumber {
+		head := api.chain.CurrentHeader()
+		if head == nil {
+			return nil, errUnknownBlock
+		}
+		return head, nil
+	} else if *number == rpc.EarliestBlockNumber {
+		return api.chain.GetHeaderByNumber(0), errUnknownBlock
+	}
+
+	return api.chain.GetHeaderByNumber(uint64(*number)), nil
```

Comment:
Shouldn't this check the value and return `nil, errUnknownBlock` if it's `nil`?  Sounds like this function will need three nil checks.  Maybe set the result to a variable and then at the end do a single check against nil and then return according to that?